compendium/DeclarativeServices/test/TestUtils.cpp

```diff
@@ -275,4 +275,43 @@ namespace test
 #endif
     }
 
+    AsyncWorkServiceThreadPool::AsyncWorkServiceThreadPool(int nThreads) : cppmicroservices::async::AsyncWorkService(), threadpool(std::make_shared<boost::asio::thread_pool>(nThreads))
+    {
+    }
+
+    AsyncWorkServiceThreadPool::~AsyncWorkServiceThreadPool()
+    {
+        try
+        {
+            if (threadpool)
+            {
+                try
+                {
+                    threadpool->join();
+                }
+                catch (...)
+                {
+                    //
+                }
+            }
+        }
+        catch (...)
+        {
+            //
+        }
+    }
+
+    void
+    AsyncWorkServiceThreadPool::post(std::packaged_task<void()>&& task)
+    {
+        using Sig = void();
+        using Result = boost::asio::async_result<decltype(task), Sig>;
+        using Handler = typename Result::completion_handler_type;
+
+        Handler handler{std::forward<decltype(task)>(task)};
```

Comment:
`std::forward` makes sense only for forwarding references (i.e. r-value reference to a templated type). In this case, the type isn't a template and you have a regular r-value, so you can just use `std::move` ?

Reply:
I can fix this for sure. This was just moved from the testAsyncWorkService file so I hadn't thought too much about it.